tcpMaxConn.go

```diff
@@ -1,34 +1,91 @@
 package main
 
 import (
+	"bufio"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"os"
+	"strconv"
+	"strings"
+
 	"github.com/dachad/check-max-tcp-connections/mtcpclient"
 	"github.com/spf13/pflag"
 )
 
 func main() {
-	hostPtr := pflag.StringP("host", "h", "localhost", "Host you want to open tcp connections against")
+	hostPtr := pflag.StringP("host", "h", "", "Host you want to open tcp connections against")
 	// according to https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers, you are probably not using this
-	portPtr := pflag.IntP("port", "p", 9998, "Port you want to open tcp connections against")
+	portPtr := pflag.StringP("port", "p", "", "Port you want to open tcp connections against")
 	numberConnectionsPtr := pflag.IntP("connections", "c", 100, "Number of connections you want to open")
 	delayPtr := pflag.IntP("sleep", "s", 10, "Time you want to sleep between connections, in ms")
 	debugPtr := pflag.BoolP("debug", "d", false, "Print debugging information to the standard error")
 	reportingIntervalPtr := pflag.IntP("interval", "i", 1, "Interval, in seconds, between stats updates")
+	assumeyesPtr := pflag.BoolP("assume-yes", "y", false, "Force execution without asking for confirmation")
 	pflag.Parse()
 
+	// Target host and port are mandatory to run the TCP check
+	if *hostPtr == "" || *portPtr == "" {
+		pflag.PrintDefaults()
+		os.Exit(1)
+	}
+	port, _ := strconv.Atoi(*portPtr)
+
 	var debugOut io.Writer = ioutil.Discard
 	if *debugPtr {
 		debugOut = os.Stderr
 	}
 
-	connStatusCh := mtcpclient.StartReportingLogic(*numberConnectionsPtr, *reportingIntervalPtr)
+	if *assumeyesPtr == false {
+		*assumeyesPtr = askForUserConfirmation(*hostPtr, *portPtr, *numberConnectionsPtr)
```

Comment:
reusing vars like this is ugly! Why don't just group the ifs in something like this:
```
if *assumeyesPtr ||  askForUserConfirmation(*hostPtr, *portPtr, *numberConnectionsPtr) {
```
It should only call the function if assumeyesPtr is false...